Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingFeatureProcessor.cpp

```diff
@@ -77,30 +77,29 @@ namespace AZ
             AZ_Assert(m_rayTracingMaterialSrg, "Failed to create RayTracingMaterialSrg");
         }
 
-        void RayTracingFeatureProcessor::SetMesh(const ObjectId objectId, const AZ::Data::AssetId& assetId, const SubMeshVector& subMeshes)
+        void RayTracingFeatureProcessor::AddMesh(const AZ::Uuid& uuid, const AZ::Data::AssetId& assetId, const SubMeshVector& subMeshes, const AZ::Transform& transform, const AZ::Vector3& nonUniformScale)
         {
             if (!m_rayTracingEnabled)
             {
                 return;
             }
 
             RHI::Ptr<RHI::Device> device = RHI::RHISystemInterface::Get()->GetDevice();
-            uint32_t objectIndex = objectId.GetIndex();
 
             // lock the mutex to protect the mesh and BLAS lists
             AZStd::unique_lock<AZStd::mutex> lock(m_mutex);
 
             // check to see if we already have this mesh
-            MeshMap::iterator itMesh = m_meshes.find(objectIndex);
+            MeshMap::iterator itMesh = m_meshes.find(uuid);
             if (itMesh != m_meshes.end())
             {
-                AZ_Assert(false, "SetMesh called on an existing Mesh objectId, call RemoveMesh first");
+                AZ_Assert(false, "AddMesh called on an existing Mesh, call RemoveMesh first");
                 return;
             }
 
             // add the mesh
-            m_meshes.insert(AZStd::make_pair(objectIndex, Mesh{ assetId }));
-            Mesh& mesh = m_meshes[objectIndex];
+            m_meshes.insert(AZStd::make_pair(uuid, Mesh{ assetId }));
```

Comment:
Nit: I think `insert` returns an iterator/inserted-state pair you can use to retrieve the mesh you just added to avoid the extra lookup on the next line.